src/components/generic/TimerComps.tsx

```diff
@@ -0,0 +1,289 @@
+};
+
+/**
+ * Props for the TimerTypeSelect component.
+ */
+type selectTimerProps = {
+    timerTypes: {
+        title: string;
+    }[];
+    selected: number;
+    setSelected: (value: number) => void;
+};
+
+/*
+ * Component to handle the selection of the time of timer to add.
+ */
+export const TimerTypeSelect = ({ timerTypes, selected, setSelected }: selectTimerProps) => {
+    return (
+        <div className="w-64">
+            <label htmlFor="timerType" className="px-4 font-bold">
+                Timer Type
+            </label>
+            <select
+                id="timerType"
+                className="bg-gray-900 text-white rounded-md px-3 py-3 text-sm font-medium"
+                value={selected}
+                onChange={event => {
+                    setSelected(Number.parseInt(event.target.value));
+                }}
+            >
+                {timerTypes.map((timer, index) => (
+                    <option key={timer.title} value={index}>
+                        {timer.title}
+                    </option>
+                ))}
+            </select>
+        </div>
+    );
+};
+
+export const TimerInputsContainer = ({ children }: { children: React.ReactNode }) => {
+    return (
+        <div className="flex flex-wrap w-full">
+            <div className="w-full">{children}</div>
+        </div>
+    );
+};
+
+export type TimerInputComponentProps = {
+    label: string;
+    id: string;
+    defaultValue: number;
+    min: number;
+    max: number;
+    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+export const TimerInputComponent = ({ label, id, defaultValue, min, max, onChange }: TimerInputComponentProps) => {
+    return (
+        <>
+            <div className="flex items-center ">
+                <label className="text-white font-bold mb-1 md:mb-0 whitespace-nowrap mr-2" htmlFor={id}>
+                    {label}
+                </label>
+            </div>
+            <div className="flex items-center col-span-3 w-64">
+                <input
+                    className="appearance-none border-2 border-gray-200 rounded w-full py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-gray-800"
+                    id={id}
+                    type="number"
+                    defaultValue={defaultValue}
+                    min={min}
+                    max={max}
+                    onChange={onChange}
+                />
+            </div>
+        </>
+    );
+};
+
+/**
+ * Props for the TimerInput component.
+ */
+type TimerInputProps = {
+    setDuration: (value: number) => void;
+    label: string;
+};
+
+/**
+ * Component for the input of the time of the countdown or Stopwatch timers.
+ */
+export const TimerInput = ({ setDuration, label }: TimerInputProps) => {
+    return (
+        <TimerInputsContainer>
+            <TimerInputComponent
+                label={label}
```

Comment:
the UI should work in units that make sense to the user, not to the developer. Users of this application will think in minutes, seconds, not in milliseconds.